### PR TITLE
Auto check binding

### PR DESCRIPTION
**Pull request by yuema137**
Related to https://github.com/XENONnT/utilix/pull/110 and the idea is basically the same
This PR does the following major changes:
- Combine several shell scripts written for each server into one, as they are basically the same except for some different settings. So now we only have one core script, which is `start_notebook.sh`
- Instead of hard-coded locations, we put all possible locations for images so that the program can search from the list
- Add an auto-check for the bindings. We only keep usable ones and throw a warning to the users for the invalid ones.
I have tested the updated functions on midway2, midway3 and dali, and they all work well.

### CODE REVIEW

**Comment by ghusheng** [comment]
Hi Yue! Than you for the PR! Could you also add `--bind /project/lgrandi/xenonnt/dali:/dali` to midway3, otherwise the cutax may be missing. Thanks!

**Comment by GiovanniVolta** [comment]
Hey @yuema137, it does not work for me. The variable $PARTITION in start_notebook.sh has no value

File: start_jupyter.py [hunk 1]
```diff
@@ -249,7 +241,7 @@ def main():
         batch_job = JOB_HEADER + \
                     "{env_starter}/{script} " \
                     "{s_container} {jupyter} {nbook_dir}".format(env_starter=ENVSTARTER_PATH,
```
Comment (GiovanniVolta) [inline start_jupyter.py:243]: It should be `{s_container} {jupyter} {nbook_dir} {partition}` no?

**Review by GiovanniVolta — CHANGES_REQUESTED** [review]

File: start_notebook.sh [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+fi
+
+echo "Using singularity image: ${CONTAINER}"
+
+PORT=$((15000 + (RANDOM %= 5000)))
+
+if [[ "$PARTITION" == "lgrandi" || "$PARTITION" == "build" || "$PARTITION" == "caslake" ]]; then
+  SINGULARITY_CACHEDIR=/scratch/midway3/$USER/singularity_cache
+  SSH_HOST="midway3.rcc.uchicago.edu"
+  BIND_OPTS=("--bind /project2" "--bind /scratch/midway3/$USER" "--bind /project/lgrandi")
```
Comment (GiovanniVolta) [inline start_notebook.sh:41]: Here you missed "--bind /project/lgrandi/xenonnt/dali:/dali"

**Review by GiovanniVolta — CHANGES_REQUESTED** [review]

File: start_notebook.sh (hunk 2 above)
Comment (yuema137) [reply start_notebook.sh]: Resolved. Thanks for spotting!

**Comment by yuema137** [comment]
@GiovanniVolta @ghusheng Thanks for the catches! It's already updated, and it should be good now. Please let me know if there are any missing edge cases

File: start_notebook.sh [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    OR replace "$ipnip" in the address below with "localhost" and copy
+    to your local browser.
+    " 2>&1
+
+jupyter ${JUPYTER_TYPE} --no-browser --port=$PORT --ip=\$JUP_HOST --notebook-dir ${NOTEBOOK_DIR} 2>&1
+EOF
+chmod +x $INNER
+
+if [[ "$PARTITION" == "dali" ]]; then
```
Comment (rsriya) [inline start_notebook.sh:79]: Please add following for midway3 in the same way 
`export XENON_CONFIG=/project/lgrandi/xenonnt/xenon.config`

**Review by rsriya — CHANGES_REQUESTED** [review]

File: start_notebook_dali.sh [hunk 4]
```diff
@@ -1,58 +0,0 @@
...
-  JUPYTER_TYPE='lab'
-fi
-
-echo "Using singularity image: ${CONTAINER}"
-
-PORT=$(( 15000 + (RANDOM %= 5000) ))
-SINGULARITY_CACHEDIR=/dali/lgrandi/$USER/singularity_cache
-
-# script to run inside container
-INNER=/dali/lgrandi/$USER/.singularity_inner
```
Comment (rsriya) [inline start_notebook_dali.sh:31]: Why was this whole path instead of just being .singularity_inner?

**Review by GiovanniVolta — APPROVED** [review]

File: start_notebook_dali.sh (hunk 4 above)
Comment (yuema137) [reply start_notebook_dali.sh:31]: I don't know why people wrote it differently for dali. But I think it's better to be consistent here. As people can only start notebook with the dali partition on dali, setting INNER by PWD should be the best practice.

File: start_notebook.sh (hunk 3 above)
Comment (yuema137) [reply start_notebook.sh:79]: added

**Review by rsriya — APPROVED** [review]

**Pull request merged**